pkg/migrate/azure/privatedns.go

```diff
@@ -0,0 +1,744 @@
+	var mxRecords []azprivatedns.MxRecord = nil
+
+	if legacyMxRecords != nil {
+		mxRecords = make([]azprivatedns.MxRecord, len(*legacyMxRecords))
+		for _, legacyMxRecord := range *legacyMxRecords {
+			mxRecord := legacyMxRecordToPrivate(&legacyMxRecord)
+			if mxRecord != nil {
+				mxRecords = append(mxRecords, *mxRecord)
+			}
+		}
+	}
+
+	return &mxRecords
+}
+
+// convert an array of legacy A records to an array of private A records
+func legacyARecordsToPrivate(legacyARecords *[]azdns.ARecord) *[]azprivatedns.ARecord {
+	var aRecords []azprivatedns.ARecord = nil
+
+	if legacyARecords != nil {
+		for _, legacyARecord := range *legacyARecords {
+			aRecord := legacyARecordToPrivate(&legacyARecord)
+			if aRecord != nil {
+				aRecords = append(aRecords, *aRecord)
+			}
+
+		}
+	}
+
+	return &aRecords
+}
+
+// convert an array of legacy AAAA records to an array of private AAAA records
+func legacyAaaaRecordsToPrivate(legacyAaaaRecords *[]azdns.AaaaRecord) *[]azprivatedns.AaaaRecord {
+	var aaaaRecords []azprivatedns.AaaaRecord = nil
+
+	if legacyAaaaRecords != nil {
+		for _, legacyAaaaRecord := range *legacyAaaaRecords {
+			aaaaRecord := legacyAaaaRecordToPrivate(&legacyAaaaRecord)
+			if aaaaRecord != nil {
+				aaaaRecords = append(aaaaRecords, *aaaaRecord)
+			}
+		}
+	}
+
+	return &aaaaRecords
+}
+
+// convert an array of legacy PTR records to an array of private PTR records
+func legacyPtrRecordsToPrivate(legacyPtrRecords *[]azdns.PtrRecord) *[]azprivatedns.PtrRecord {
+	var ptrRecords []azprivatedns.PtrRecord = nil
+
+	if legacyPtrRecords != nil {
+		for _, legacyPtrRecord := range *legacyPtrRecords {
+			ptrRecord := legacyPtrRecordToPrivate(&legacyPtrRecord)
+			if ptrRecord != nil {
+				ptrRecords = append(ptrRecords, *ptrRecord)
+			}
+		}
+	}
+
+	return &ptrRecords
+}
+
+// convert an array of legacy SRV records to an array of private SRV records
+func legacySrvRecordsToPrivate(legacySrvRecords *[]azdns.SrvRecord) *[]azprivatedns.SrvRecord {
+	var srvRecords []azprivatedns.SrvRecord = nil
+
+	if legacySrvRecords != nil {
+		for _, legacySrvRecord := range *legacySrvRecords {
+			srvRecord := legacySrvRecordToPrivate(&legacySrvRecord)
+			if srvRecord != nil {
+				srvRecords = append(srvRecords, *srvRecord)
+			}
+		}
+	}
+
+	return &srvRecords
+}
+
+// convert an array of legacy TXT records to an array of private TXT records
+func legacyTxtRecordsToPrivate(legacyTxtRecords *[]azdns.TxtRecord) *[]azprivatedns.TxtRecord {
+	var txtRecords []azprivatedns.TxtRecord = nil
+
+	if legacyTxtRecords != nil {
+		for _, legacyTxtRecord := range *legacyTxtRecords {
+			txtRecord := legacyTxtRecordToPrivate(&legacyTxtRecord)
+			if txtRecord != nil {
+				txtRecords = append(txtRecords, *txtRecord)
+			}
+		}
+	}
+
+	return &txtRecords
+}
+
+// Transforms a legacy zone to a private zone
+func (client *privateDNSClient) MigrateLegacyZone(legacyDNSZoneInfo *legacyDNSZoneInfo, link bool) error {
```

Comment:
why is this function public?

Reply:
also why are we calling Migrate on the new zone type client? isn't that reverse? we should be calling migrate on the legacy zone client??

Reply:
I've made the function private.